codec-http2/src/main/java/io/netty/handler/codec/http2/internal/hpack/HuffmanDecoder.java

```diff
@@ -213,4 +138,114 @@ private static void insert(Node root, int symbol, int code, byte length) {
             current.children[i] = terminal;
         }
     }
+
+    private static final class DecoderProcessor implements ByteProcessor {
+        // TODO: Is this a good size ?
+        private static final int INITIAL_SIZE = 64;
+        private byte[] bytes = new byte[INITIAL_SIZE];
+        private int index;
+
+        private Node node = ROOT;
+        private int current;
+        private int currentBits;
+        private int symbolBits;
+
+        void reset() {
+            node = ROOT;
+            current = 0;
+            currentBits = 0;
+            symbolBits = 0;
+            bytes = new byte[INITIAL_SIZE];
```

Comment:
@Scottmitch @nmittler another thing we could do is allocate a byte[] for the lifetime of DecoderProcessor and just "slice" out `ArrayString` instances via and offset and length. Not sure if this would be better or worse... WDYT ?